File: 0020-valid-parentheses/0020-valid-parentheses.c
bool isValid(char* s) {
    
    int len = strlen(s);
    char stack[len + 1];
    char* head = stack;
    char* ptr = stack;
    int k = ptr == head;
    for(int i = 0; i < len; i++){
        char ch = s[i];
        switch(ch){
            case ']':
                if (*ptr != '['){
                    return false;
                }
                else{
                    *ptr-- = '\0';
                }
                break;
            case '}':
                if(*ptr != '{'){
                    return false;
                }
                else {
                    *ptr-- = '\0';
                }
                break;
            case ')':
                if(*ptr != '('){
                    return false;
                }
                else{
                    *ptr-- = '\0';
                }
                break;
            case '[':
            case '{':
            case '(':
                *++ptr = ch;
                break;
        }
    }
    printf("%d\n", ptr == head);
    return ptr == head;
}

File: 0020-valid-parentheses/0020-valid-parentheses.rs
use std::vec::Vec;
impl Solution {
    pub fn is_valid(s: String) -> bool {
        let mut stack:Vec<char> = Vec::new();
        for c in s.chars(){
            if c == '{' {
                stack.push('}');
            }
            else if c == '(' {
                stack.push(')');
            }
            else if c == '[' {
                stack.push(']');
            }
            else {
                if stack.len() == 0 { return false; }
                if stack.pop() != Some(c) { return false; }
            }
        }
            return stack.len() == 0;
    }
}